src/main/java/seedu/duke/userinterface/InputParser.java

```diff
@@ -91,35 +104,48 @@ private static boolean correctTimeFormat(String by) {
     }
 
     /**
-     * Extracts the parameters given by the user when selecting in notebook mode.
+     * Parses user's input to extract notebook title, section title or page number whenever applicable.
      *
      * @param argument contains notebook title, section title or/and page number.
      * @param appState is the state of the application.
      */
     public void extractParams(String argument, AppState appState)
-            throws InvalidCommandException, InvalidNotebookException, InvalidPageException, InvalidSectionException {
-        if (argument.startsWith(NOTEBOOK_DELIMITER) && appState.getAppMode() == AppMode.NOTEBOOK_SHELF) {
+            throws InvalidNotebookException, InvalidSectionException, InvalidPageException, InvalidCommandException,
+            NotebookOutOfBoundsException {
+        if (argument.startsWith(NOTEBOOK_DELIMITER)) {
             extractNotebookParams(argument, appState);
-        } else if (argument.startsWith(SECTION_DELIMITER) && appState.getAppMode() == AppMode.NOTEBOOK_BOOK) {
+        } else if (argument.startsWith(SECTION_DELIMITER)) {
             extractSectionParams(argument, appState);
-        } else if (argument.startsWith(PAGE_DELIMITER) && appState.getAppMode() == AppMode.NOTEBOOK_SECTION) {
+        } else if (argument.startsWith(PAGE_DELIMITER)) {
             Section section = appState.getCurrentSection();
             int pageNum = parsePageNumber(argument);
             section.getPage(pageNum);
+        } else if (argument.startsWith(SHOW_ALL)) {
+            appState.setAppMode(AppMode.NOTEBOOK_SHELF);
         } else {
             throw new InvalidCommandException(argument);
         }
     }
 
+    /**
+     * Extracts the notebook title, as well as the section title and page number, if provided.
+     *
+     * @param argument is the user's input.
+     * @param appState is the current mode the user is in.
+     * @throws InvalidNotebookException when the notebook title input by the user does not exist.
+     * @throws InvalidSectionException  when the section title input by the user does not exist.
+     * @throws InvalidPageException     when the page number input by the user does not exist.
+     */
     public void extractNotebookParams(String argument, AppState appState)
-            throws InvalidNotebookException, InvalidSectionException, InvalidPageException {
+            throws InvalidNotebookException, InvalidSectionException, InvalidPageException,
+            NotebookOutOfBoundsException {
         Notebook notebook;
         Section section = null;
         String notebookTitle = parseNotebookTitle(argument);
         NotebookShelf notebookShelf = appState.getCurrentBookShelf();
         int notebookIndex = notebookShelf.findNotebook(notebookTitle);
         if (notebookIndex == -1) {
```

Comment:
We should try to remove these -1 as it's a little confusing without any context

Reply:
What do you suggest changing it to? xD Or you mean I should just replace that line with sth like `if (notebookIndex < 0)` ?

Reply:
Oh like what's the significance of -1? Or its supposed to perform like `if (notebookIndex < 0)` ?

Reply:
ooooooh the -1 is thrown when the notebook/section/page cannot be found (when the user selects a notebook/section/page that doesn't exist)... so you don't like that the -1 is being thrown is it ?